pkg/adapter/incoming.go

```diff
@@ -86,6 +121,97 @@
 	return true, repo, nil
 }
 
+func (l *listener) getAndUpdateInstallationID(ctx context.Context, req *http.Request, run *params.Run, repo *v1alpha1.Repository) error {
+	JWTtoken, err := GenerateJWT(ctx, run)
+	if err != nil {
+		return err
+	}
+
+	enterpriseURL := req.Header.Get("X-GitHub-Enterprise-Host")
+	if enterpriseURL != "" {
+		apiURL = enterpriseURL
+	}
+
+	res, err := getResponse(ctx, http.MethodGet, installationURL, JWTtoken, run)
```

Comment:
go-github should have all of this ie https://github.com/chmouel/pipelines-as-code/blob/rework-documentation/vendor/github.com/google/go-github/v48/github/apps.go#L181 can we remove this code and use the library code instead ?

Reply:
I tried to use library code  but client is getting nil https://github.com/chmouel/pipelines-as-code/blob/rework-documentation/vendor/github.com/google/go-github/v48/github/apps.go#L187 because of no installtion ID which we use  in parse_payload something like this https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/provider/github/parse_payload.go#L47